### PR TITLE
ENG-1159 Store deploy args in S3

### PR DESCRIPTION
**Pull request by ryansch**
We've finally hit the 8K limit for container overrides with the ECS API. This PR moves the deploy configuration to S3.

I've also added a _ton_ of cleanup and updates:
- CI
- Deskfile
- ECR Credential Helper 0.5.0: https://github.com/awslabs/amazon-ecr-credential-helper/blob/main/CHANGELOG.md
- Buildkit 0.8.3: https://github.com/moby/buildkit/releases
- buildx 0.5.1: https://github.com/docker/buildx/releases
- bundler 2.2.21
- ruby 2.7.3
- Dedicated `staging` container image for testing upcoming shipitron changes in prod

This PR also introduces a new global config file located at `~/.config/shipitron/config.yml`. I'll provide you with the contents.

This will be version 1.4.0.

### CODE REVIEW

File: spec/unit/git_info_spec.rb [hunk 1]
```diff
@@ -23,7 +23,7 @@
       expect(git_info.name).to eq 'Ryan Schlesinger'
       expect(git_info.summary).to eq 'Add bar'
       expect(git_info.timestamp).to eq '1601476924'
-      expect(git_info.branch).to eq 'main'
```
Comment (morganick) [inline spec/unit/git_info_spec.rb:26]: ?

Comment (ryansch) [reply spec/unit/git_info_spec.rb:26]: It's burned into the test fixture.

**Comment by morganick** [comment]
@ryansch I'm going to try to run the specs here locally. Then I'll give you the 👍

**Review by morganick — APPROVED** [review]
Works as expected.

**Pull request merged**